File: Python/variable_assign.py
import time

ONE_BILLION = 1000000000
NUM_TRIALS = 50
cumulative = 0

for t in range(NUM_TRIALS):
    startTime = time.time_ns()
    x = 0
    while x < ONE_BILLION:
        x += 1
    endTime = time.time_ns()
    difference = endTime - startTime
    cumulative += difference
    seconds = difference / ONE_BILLION
    print(f"Time to process: {seconds} seconds")
    time.sleep(0.0003)

seconds = cumulative / ONE_BILLION / NUM_TRIALS
print(f"Average time to process: {seconds} seconds")
    

File: Rust/variable_assign.rs
use std::time::Instant;
use std::thread::sleep;

fn main() {
    const ONE_BILLION_D : f64 = 1000000000.0;
    const ONE_BILLION : u64 = 1000000000;
    const ONE_MILLION : u32 = 1000000;
    const NUM_TRIALS : u16 = 50;

    let mut cumulative : f64 = 0.0;
    for _i in 0..(NUM_TRIALS-1) {
        let start_time = Instant::now();
        let mut x : u64 = 0;
        while x < ONE_BILLION {
            x += 1;
        }
        let end_time = Instant::now();
        let difference = (end_time - start_time).as_secs() * ONE_BILLION +  (end_time - start_time).subsec_nanos() as u64;
        let seconds : f64 = difference as f64 / ONE_BILLION_D;
        cumulative += seconds;
        println!("Time to process: {} seconds", seconds);
    }

    let seconds = cumulative / NUM_TRIALS as f64;
    println!("Average time to process: {} seconds", seconds);

}